### PR TITLE
sync_diff: fix hidden column

### LINKED ISSUE #713: sync_diff: can't check the upstream with hidden column
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do?
If possible, provide a recipe for reproducing the error.

#### upstream

```
CREATE TABLE `t` (
  `pk` bigint(20) NOT NULL AUTO_INCREMENT,
  `j` json DEFAULT NULL,
  `i` int(11) DEFAULT NULL,
  `c` char(64) DEFAULT NULL,
  PRIMARY KEY (`pk`) /*T![clustered_index] CLUSTERED */,
  KEY `j` ((cast(json_extract(`j`, _utf8mb4'$.number[*]') as signed array)))
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin AUTO_INCREMENT=30001
```

the upstream has a hidden column `_V$_j_0`

```log
[2023/03/17 13:07:07.517 +08:00] [DEBUG] [tidb.go:189] ["select data"] [sql="SELECT /*!40001 SQL_NO_CACHE */ `pk`, `j`, `i`, `c`, `_V$_j_0` FROM `test`.`t` WHERE ((TRUE) AND (TRUE)) ORDER BY `pk`"] [args=null]
```

2. What did you expect to see?

skip the hidden column

3. What did you see instead?

4. What version of TiDB are you using (`tidb-server -V` or run `select tidb_version();` on TiDB)?

5. which tool are you using?

6. what versionof tool are you using (`pump -V` or `tidb-lightning -V` or `syncer -V`)?

### PR DESCRIPTION
**Pull request by okJiang**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb-tools/issues/713

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change
 - Has persistent data change

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation
 - Need to be included in the release note

### CODE REVIEW

**Comment by lance6716** [comment]
@okJiang Can you add a test, or manually test it?

**Comment by okJiang** [comment]
> @okJiang Can you add a test, or manually test it?

This PR has not yet solved this issue, and it has been manually tested. 

@Leavrth  will take a look for this in Friday

**Review by YangKeao — APPROVED** [review]
LGTM

**Comment by YangKeao** [comment]
I think https://github.com/pingcap/tidb-tools/pull/715 is part of it, so you need to resolve the conflict.

**Comment by Leavrth** [comment]
the index having expression is not support for `binary search`, `bucket split` and `random split` features in sync-diff-inspector.
So ignore this kind of the index when build splitter.

(However, sync-diff-inspector doesn't compare the index kvs.

File: tests/sync_diff_inspector/expression/run.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+rm -rf $OUT_DIR
+rm -rf $FIX_DIR
+mkdir -p $OUT_DIR
+mkdir -p $FIX_DIR
+
+for port in 4000 4001; do
+  mysql -uroot -h 127.0.0.1 -P $port -e "create database if not exists expression_test;"
+  mysql -uroot -h 127.0.0.1 -P $port -e "create table expression_test.diff(`a``;sad` int, id int);"
+  mysql -uroot -h 127.0.0.1 -P $port -e "alter table expression_test.diff add index i1((`a``;sad` + 1 + `a``;sad`));"
+  mysql -uroot -h 127.0.0.1 -P $port -e "insert into expression_test.diff values (1,1),(2,2),(3,3);"
```
Comment (Leavrth) [inline tests/sync_diff_inspector/expression/run.sh]:
```suggestion
  mysql -uroot -h 127.0.0.1 -P $port -e "create table expression_test.diff(\`a\`\`;sad\` int, id int);"
  mysql -uroot -h 127.0.0.1 -P $port -e "alter table expression_test.diff add index i1((\`a\`\`;sad\` + 1 + \`a\`\`;sad\`));"
  mysql -uroot -h 127.0.0.1 -P $port -e "insert into expression_test.diff values (1,1),(2,2),(3,3);"
```

**Review by Leavrth — APPROVED** [review]

**Pull request merged**